qt/applications/workbench/workbench/plugins/workspacewidget.py

```diff
@@ -121,8 +121,17 @@ def _do_show_instrument(self, names):
         :param names: A list of workspace names
         """
         for ws in self._ads.retrieveWorkspaces(names, unrollGroups=True):
-            presenter = InstrumentViewPresenter(ws, parent=self)
-            presenter.show_view()
+            if ws.getInstrument().getName():
+                try:
+                    presenter = InstrumentViewPresenter(ws, parent=self)
+                    presenter.show_view()
+                except Exception as exception:
+                    logger.warning("Could not show instrument for workspace "
+                                   "'{}':\n{}.\n".format(ws.name(), exception))
+            else:
+                logger.warning("Could not show instrument for workspace '{}':"
+                               "\nNo instrument view available.\n"
```

Comment:
This is really minor but could we just say "No instrument available" as it's slightly more meaningful to users.